test/http/write.cpp

```diff
@@ -87,22 +124,333 @@ class write_test : public beast::detail::unit_test::suite
         };
     };
 
+    struct fail_body
+    {
+        class writer;
+
+        class value_type
+        {
+            friend class writer;
+
+            std::string s_;
+            test::fail_counter& fc_;
+            boost::asio::io_service& ios_;
+
+        public:
+            value_type(test::fail_counter& fc,
+                    boost::asio::io_service& ios)
+                : fc_(fc)
+                , ios_(ios)
+            {
+            }
+
+            boost::asio::io_service&
+            get_io_service() const
+            {
+                return ios_;
+            }
+
+            value_type&
+            operator=(std::string s)
+            {
+                s_ = std::move(s);
+                return *this;
+            }
+        };
+
+        class writer
+        {
+            std::size_t n_ = 0;
+            value_type const& body_;
+            bool suspend_ = false;
+            enable_yield_to yt_;
```

Comment:
Hey, I provide the hammer. You're free to hold it any way that you want. Including using your feet.